app/main.py

```diff
@@ -1,6 +1,15 @@
 from typing import Callable
+_CACHE = {}
 
 
 def cache(func: Callable) -> Callable:
-    # Write your code here
-    pass
+    def wrapper(*args: int) -> int:
+        if func not in _CACHE:
+            _CACHE[func] = {}
+        if args not in _CACHE[func]:
+            _CACHE[func][args] = func(*args)
+            print("Calculating new result")
+        else:
+            print("Getting from cache")
+        return _CACHE[func][args]
```

Comment:
There is no need to store info about function in cache, you are working harder, and we need to think. Why not storing cache inside outer function? Then each function that will be decorated will have its own cache, and you can only store args of that function to check if we already worked with that data.
Also, this approach will help you to reduce and simplify your code